repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -1767,14 +1767,23 @@ private AtlasEdge getEdgeUsingRelationship(AttributeMutationContext ctx, EntityM
                     toVertex   = attributeVertex;
                 }
 
-                AtlasEdge edge = relationshipStore.getRelationship(fromVertex, toVertex, new AtlasRelationship(relationshipName));
+                AtlasEdge edge = null;
 
-                if (edge != null && getStatus(edge) != DELETED) {
-                    ret = edge;
-                }
+                if (createEdge) {
+                    edge = relationshipStore.getOrCreate(fromVertex, toVertex, new AtlasRelationship(relationshipName));
+                    boolean isCreated = graphHelper.getCreatedTime(edge) == RequestContext.get().getRequestTime();
```

Comment:
dint get this `boolean isCreated = graphHelper.getCreatedTime(edge) == RequestContext.get().getRequestTime();
` why check on time and will the time always a match ?

Reply:
Hi Naveen ,
I got this from legacy code 
https://github.com/atlanhq/atlas-metastore/blob/a6a7d78d176913e3c5c84881a2fce94c5d675df7/repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java#L1672C114-L1673C1